pkgs/applications/audio/openutau/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, buildDotnetModule
+, fetchFromGitHub
+, dotnetCorePackages
+, dbus
+, fontconfig
+, libICE
+, libSM
+, libX11
+, portaudio
+}:
+
+buildDotnetModule rec {
+  pname = "OpenUtau";
+  version = "0.1.158";
+
+  src = fetchFromGitHub {
+    owner = "stakira";
+    repo = "OpenUtau";
+    rev = "build/${version}";
+    hash = "sha256-/+hlL2sj/juzWrDcb5dELp8Zdg688XK8OnjKz20rx/M=";
+  };
+
+  dotnet-sdk = dotnetCorePackages.sdk_7_0;
+  dotnet-runtime = dotnetCorePackages.runtime_7_0;
+
+  projectFile = "OpenUtau.sln";
+  nugetDeps = ./deps.nix;
+
+  executables = [ "OpenUtau" ];
+
+  runtimeDeps = [
+    dbus
+    fontconfig
+    libICE
+    libSM
+    libX11
+    portaudio
+  ];
+
+  dotnetInstallFlags = [ "-p:PublishReadyToRun=false" ];
+
+  # socket cannot bind to localhost on darwin for tests
+  doCheck = !stdenv.isDarwin;
+
+  # needed until upstream bumps to dotnet 7
+  postPatch = ''
+    substituteInPlace OpenUtau/OpenUtau.csproj OpenUtau.Test/OpenUtau.Test.csproj --replace \
+      "<TargetFramework>net6.0</TargetFramework>" \
+      "<TargetFramework>net7.0</TargetFramework>"
+  '';
+
+  # need to make sure proprietary worldline resampler is copied
+  postInstall = let
+    runtime = if (stdenv.isLinux && stdenv.isx86_64) then "linux-x64"
+         else if (stdenv.isLinux && stdenv.isAarch64) then "linux-arm64"
+         else if stdenv.isDarwin then "osx"
+         else null;
+  in lib.optionalString (runtime != null) ''
+    cp runtimes/${runtime}/native/libworldline${stdenv.hostPlatform.extensions.sharedLibrary} $out/lib/OpenUtau/
+  '';
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
+    description = "Open source singing synthesis platform and UTAU successor";
+    homepage = "http://www.openutau.com/";
+    sourceProvenance = with sourceTypes; [
+      fromSource
+      # deps
+      binaryBytecode
+      # some deps and worldline resampler
+      binaryNativeCode
+    ];
```

Comment:
I didn't know this exists before just looking this up, so I cant really say a lot about it :see_no_evil:

Reply:
Which part, `meta.sourceProvenance` or one of the values?

Reply:
Yes `meta.sourceProvenance`, the lines marked above.

Reply:
It's to be able to programmtically differentiate project we packagw that use prebuilt artifacts rather than only using stuff built from source (and different whether it's prebuilt bytecode (like for JVM or .NET) or native code)